iotdb-core/datanode/src/main/java/org/apache/iotdb/db/subscription/broker/SubscriptionBroker.java

```diff
@@ -162,8 +162,30 @@ public void unbindPrefetchingQueue(final String topicName) {
       LOGGER.warn("Subscription: prefetching queue bound to topic [{}] does not exist", topicName);
       return;
     }
+
     // clean up uncommitted events
     prefetchingQueue.cleanup();
+    prefetchingQueue.markCompleted();
+
+    // we remove this prefetching queue and deregister metrics when consumers dropping related
+    // subscription...
+  }
+
+  public void removePrefetchingQueue(final String topicName) {
+    final SubscriptionPrefetchingQueue prefetchingQueue =
+        topicNameToPrefetchingQueue.get(topicName);
+    if (Objects.isNull(prefetchingQueue)) {
+      LOGGER.warn("Subscription: prefetching queue bound to topic [{}] does not exist", topicName);
+      return;
+    }
+
+    if (!prefetchingQueue.isCompleted()) {
```

Comment:
It seems that the "completed" is marked when the pipe is dropped. However in the normal scenarios (namely the unsubscriptions is manually triggered by the client) the "drop pipe" is executed after updating consumer group meta. That is, the "isCompleted" may be false when a topic's last subscription is removed.